larray/core/array.py

```diff
@@ -6539,26 +6545,39 @@ def growth_rate(self, axis=-1, d=1, label='upper'):
 
         Examples
         --------
-        >>> sex = Axis('sex=M,F')
-        >>> year = Axis(range(2016, 2020), 'year')
-        >>> a = LArray([[1.0, 2.0, 3.0, 3.0], [2.0, 3.0, 1.5, 3.0]],
-        ...            [sex, year])
+        >>> data = [[2, 4, 5, 4, 6], [4, 6, 3, 6, 9]]
+        >>> a = LArray(data, "sex=M,F; year=2016..2020")
         >>> a
-        sex\\year  2016  2017  2018  2019
-               M   1.0   2.0   3.0   3.0
-               F   2.0   3.0   1.5   3.0
+        sex\\year  2016  2017  2018  2019  2020
+               M     2     4     5     4     6
+               F     4     6     3     6     9
         >>> a.growth_rate()
-        sex\\year  2017  2018  2019
-               M   1.0   0.5   0.0
-               F   0.5  -0.5   1.0
+        sex\\year  2017  2018  2019  2020
+               M   1.0  0.25  -0.2   0.5
+               F   0.5  -0.5   1.0   0.5
+        >>> a.growth_rate(label='lower')
+        sex\\year  2016  2017  2018  2019
+               M   1.0  0.25  -0.2   0.5
+               F   0.5  -0.5   1.0   0.5
         >>> a.growth_rate(d=2)
-        sex\\year   2018  2019
-               M    2.0   0.5
-               F  -0.25   0.0
+        sex\\year   2018  2019  2020
+               M    1.5   0.0   0.2
+               F  -0.25   0.0   2.0
+        >>> a.growth_rate('sex')
+        sex\\year  2016  2017  2018  2019  2020
+               F   1.0   0.5  -0.4   0.5   0.5
+        >>> a.growth_rate(a.year[2017:])
+        sex\\year  2018  2019  2020
+               M  0.25  -0.2   0.5
+               F  -0.5   1.0   0.5
         """
-        diff = self.diff(axis=axis, d=d, label=label)
-        axis_obj = self.axes[axis]
-        return diff / self[axis_obj.i[:-d]].drop_labels(axis)
+        if isinstance(axis, Group):
+            array = self[axis]
+            axis = Axis(axis)
+        else:
+            array = self
```

Comment:
I would rather have something like this:
```python
array = self[axis] if isinstance(axis, Group) else self
axis = array.axes[axis]
diff = array.diff(axis=axis, d=d, label=label)
return diff / array[axis.i[:-d]].drop_labels(axis)
```
so that a new axis is not created needlessly.

Reply:
line 
```python
axis = array.axes[axis]
```
fails when passing a group:
 >  File "C:\Users\ald\Projects\larray\larray\core\array.py", line 6509, in diff
 >   axis = array.axes[axis]
 >
 > File "C:\Users\ald\Projects\larray\larray\core\axis.py", line 1333, in __getitem__
 >  assert isinstance(key, basestring), type(key)

Reply:
for groups, it should be 
```python
axis = array.axes[axis.axis]
```
right?

Reply:
ok, but it should still be possible to avoid create a **new** Axis. Maybe something like:
```python
array = self[axis] if isinstance(axis, Group) else self
axis = axis.axis if isinstance(axis, Group) else axis
axis = array.axes[axis]
```

Reply:
you beat me to it :)